### ISSUE TITLE
Doc job_api

### ISSUE THREAD

**Issue by robnagler** [radiasoft/sirepo#6289]
statelessCompute, statefulCompute, etc. need clear documentation. Here's how I see things:

statelessCompute: POST simType and all inputs in request, no lib files needed (do not set libFileUri, for example). Only need access to template and the simCode.
statefulCompute:  POST simType and all inputs in request. Lib file, template, and simCode are accessed. Might be used for example for importFile, since no sid is available.
runSimulation: POST simType, computeModel, computeJobHash, simType. Run dir is needed along with lib, etc.
analysisJob: POST runSimulation is already running and runDir is available.
downloadDataFile (downloadRunFile): GET otherwise same as analysisJob. Runs as separate op.

**Comment by e-carlin** [comment]
FWIW here is some documentation https://github.com/radiasoft/sirepo/wiki/Interacting-with-Simulations-Architecture

**Comment by robnagler** [comment]
Yes, I couldn't find it, because I searched on statelessCompute. That doc should be updated, but the API (python) should have docstrings that describe what they are doing. We can use quest.Spec to give more detail and write a docstring to describe semantics. 

The [Simulations-Architecture](https://github.com/radiasoft/sirepo/wiki/Interacting-with-Simulations-Architecture) isn't quite right. statefulCompute does not get a sid so there's now way to access sirepo-data.json. Anything to do with a sim dir currently has to go through runSimulation, which requires a runDir.
